adapter_darwin.go

```diff
@@ -0,0 +1,147 @@
+var DefaultAdapter = &Adapter{
+	cm:          cbgo.NewCentralManager(nil),
+	pm:          cbgo.NewPeripheralManager(nil),
+	connectChan: make(chan cbgo.Peripheral),
+}
+
+// Enable configures the BLE stack. It must be called before any
+// Bluetooth-related calls (unless otherwise indicated).
+func (a *Adapter) Enable() error {
+	if a.poweredChan != nil {
+		return errors.New("already calling Enable function")
+	}
+
+	// wait until powered
+	a.poweredChan = make(chan struct{})
+	a.cm.SetDelegate(a)
+	select {
+	case <-a.poweredChan:
+	case <-time.NewTimer(10 * time.Second).C:
+		return errors.New("timeout enabling CentralManager")
+	}
+	a.poweredChan = nil
+
+	// wait until powered?
+	a.pm.SetDelegate(a)
+
+	return nil
+}
+
+// CentralManager delegate functions
+
+// CentralManagerDidUpdateState when central manager state updated.
+func (a *Adapter) CentralManagerDidUpdateState(cmgr cbgo.CentralManager) {
+	// powered on?
+	if cmgr.State() == cbgo.ManagerStatePoweredOn {
+		close(a.poweredChan)
+	}
+
+	// TODO: handle other state changes.
+}
+
+// DidDiscoverPeripheral when peripheral is discovered.
+func (a *Adapter) DidDiscoverPeripheral(cmgr cbgo.CentralManager, prph cbgo.Peripheral,
+	advFields cbgo.AdvFields, rssi int) {
+	if a.peripheralFoundHandler != nil {
+		sr := makeScanResult(prph, advFields, rssi)
+		a.peripheralFoundHandler(a, sr)
+	}
+}
+
+// DidConnectPeripheral when peripheral is connected.
+func (a *Adapter) DidConnectPeripheral(cmgr cbgo.CentralManager, prph cbgo.Peripheral) {
+	// Make sure our delegate gets called for events related to this
+	// peripheral.
+	//prph.SetDelegate(d)
+
+	// Unblock now that we're connected.
+	a.connectChan <- prph
+}
+
+// DidDisconnectPeripheral when peripheral is disconnected.
+func (a *Adapter) DidDisconnectPeripheral(cmgr cbgo.CentralManager, prph cbgo.Peripheral, err error) {
+}
+
+// PeripheralManager delegate functions
+
+// PeripheralManagerDidUpdateState when state updated.
+func (a *Adapter) PeripheralManagerDidUpdateState(pmgr cbgo.PeripheralManager) {
+}
+
+// DidAddService when service added.
+func (a *Adapter) DidAddService(pmgr cbgo.PeripheralManager, svc cbgo.Service, err error) {
+}
+
+// DidStartAdvertising when advertising starts.
+func (a *Adapter) DidStartAdvertising(pmgr cbgo.PeripheralManager, err error) {
+}
+
+// DidReceiveReadRequest when read request received.
+func (a *Adapter) DidReceiveReadRequest(pmgr cbgo.PeripheralManager, cbreq cbgo.ATTRequest) {
+}
+
+// DidReceiveWriteRequests when write requests received.
+func (a *Adapter) DidReceiveWriteRequests(pmgr cbgo.PeripheralManager, cbreqs []cbgo.ATTRequest) {
+}
+
+// CentralDidSubscribe when central subscribed.
+func (a *Adapter) CentralDidSubscribe(pmgr cbgo.PeripheralManager, cent cbgo.Central, cbchr cbgo.Characteristic) {
+}
+
+// CentralDidUnsubscribe when central unsubscribed.
+func (a *Adapter) CentralDidUnsubscribe(pmgr cbgo.PeripheralManager, cent cbgo.Central, chr cbgo.Characteristic) {
+}
```

Comment:
This code appears to be unused?

Reply:
This is required to use as a "Peripheral". This is why I have left it in as we will soon need it for the implementation for `Advertisement`.